packages/ra-core/src/controller/edit/useEditController.ts

```diff
@@ -134,57 +139,65 @@ export const useEditController = <
-                                          ? error
-                                          : error.message ||
-                                                'ra.notification.http_error',
-                                      {
-                                          type: 'warning',
-                                          messageArgs: {
-                                              _:
-                                                  typeof error === 'string'
-                                                      ? error
-                                                      : error && error.message
-                                                      ? error.message
-                                                      : undefined,
-                                          },
-                                      }
-                                  );
-                              },
+                                notify('ra.notification.updated', {
+                                    type: 'info',
+                                    messageArgs: { smart_count: 1 },
+                                    undoable: mutationMode === 'undoable',
+                                });
+                                redirect(redirectTo, resource, data.id, data);
+                            },
+                            onError: onErrorFromSave
+                                ? onErrorFromSave
+                                : onError
+                                ? onError
+                                : (error: Error | string) => {
+                                      notify(
+                                          typeof error === 'string'
+                                              ? error
+                                              : error.message ||
+                                                    'ra.notification.http_error',
+                                          {
+                                              type: 'warning',
+                                              messageArgs: {
+                                                  _:
+                                                      typeof error === 'string'
+                                                          ? error
+                                                          : error &&
+                                                            error.message
+                                                          ? error.message
+                                                          : undefined,
+                                              },
+                                          }
+                                      );
+                                  },
+                        }
+                    );
+                } catch (error) {
+                    if ((error as HttpError).body?.errors != null) {
+                        return (error as HttpError).body.errors;
                     }
```

Comment:
otherwise, rethrow

Reply:
Nope, this is the only type of errors we want to handle here. Otherwise, errors will be handled through the onError side effect function

Reply:
I agree with you @fzaninotto . This might even be the origin of the failing E2E test. I'll have a look.

Reply:
Okay so after verification, @djhi  is correct, both the `onError` block and the `catch` block will be executed. So we don't need to rethrow the error.

Reply:
Than please add a comment explaining why we don't rethrow and that this is not a mistake